pkg/stub/handler.go

```diff
@@ -902,68 +1003,114 @@ func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
+				// so we have an additional flag for that special case
+				logrus.Println("Initiating samples delete and marking exists false")
+				err := h.CleanUpOpenshiftNamespaceOnDelete(srcfg)
+				if err != nil {
+					return err
+				}
+				h.GoodConditionUpdate(srcfg, corev1.ConditionFalse, v1alpha1.SamplesExist)
+				logrus.Debugf("SDKUPDATE exist false update")
+				err = h.sdkwrapper.Update(srcfg)
+				if err != nil {
+					logrus.Printf("error on samplesresource update after setting exists condition to false (returning error to retry): %v", err)
+					return err
+				}
+			} else {
 				logrus.Println("Initiating finalizer processing for a SampleResource delete attempt")
 				h.RemoveFinalizer(srcfg)
-				h.CleanUpOpenshiftNamespaceOnDelete(srcfg)
-				h.GoodConditionUpdate(srcfg, corev1.ConditionFalse, v1alpha1.SamplesExist)
+				logrus.Debugf("SDKUPDATE remove finalizer update")
 				err := h.sdkwrapper.Update(srcfg)
+				if err != nil {
+					logrus.Printf("error removing samplesresource finalizer during delete (hopefully retry on return of error works): %v", err)
+					return err
+				}
 				go func() {
-					h.CreateDefaultResourceIfNeeded()
+					h.CreateDefaultResourceIfNeeded(srcfg)
 				}()
-				return err
 			}
 			return nil
 		}
 
-		doit, err := h.ProcessManagementField(srcfg)
+		doit, srcfgUpdate, err := h.ProcessManagementField(srcfg)
 		if !doit || err != nil {
-			// flush status update
-			h.sdkwrapper.Update(srcfg)
+			if err != nil || srcfgUpdate {
+				// flush status update
+				logrus.Debugf("SDKUPDATE process mgmt update")
+				h.sdkwrapper.Update(srcfg)
+			}
 			return err
 		}
 
-		// coordinate with timer's check on creating
-		// default resource ... looks at h.sampleResource,
-		// which is not set until this whole case is completed
-		h.mutex.Lock()
-		defer h.mutex.Unlock()
-
 		cm, err := h.SpecValidation(srcfg)
 		if err != nil {
 			// flush status update
+			logrus.Debugf("SDKUPDATE bad spec validation update")
 			h.sdkwrapper.Update(srcfg)
 			return err
 		}
 
-		if cm != nil {
-			changed, err := h.VariableConfigChanged(srcfg, cm)
-			if err != nil {
-				// flush status update
-				h.sdkwrapper.Update(srcfg)
-				return err
-			}
-			if !changed {
-				return nil
+		if srcfg.Spec.ManagementState == srcfg.Status.State {
+			if cm != nil {
+				changed, err := h.VariableConfigChanged(srcfg, cm)
+				if err != nil {
+					// flush status update
+					logrus.Debugf("SDKUPDATE var cfg chg err update")
+					h.sdkwrapper.Update(srcfg)
+					return err
+				}
+				// so ignore if config does not change and the samples exist and
+				// we are not in progress
+				if !changed &&
+					srcfg.ConditionTrue(v1alpha1.SamplesExist) &&
+					srcfg.ConditionFalse(v1alpha1.ImageChangesInProgress) {
+					logrus.Debugf("Handle ignoring because config the same and exists is true and in progress false")
+					return nil
+				}
 			}
 		}
 
+		srcfg.Status.State = operatorsv1alpha1api.Managed
```

Comment:
i don't see how this is guaranteed to be true?

Reply:
It corresponds with my interpretation that Removed is the same as Unmanaged and that we do not process any changes after this point.

The code above is always going to return before this point if we are Unmanaged or Removed.